src/tools/rados/rados.cc

```diff
@@ -2445,8 +2445,18 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
 
     bufferlist bl;
     if (nargs.size() > min_args) {
-      string val(nargs[min_args]);
-      bl.append(val);
+      if (omap_key.compare("order") == 0 ||
+          omap_key.compare("features") == 0 ||
+          omap_key.compare("size") == 0 ||
+          omap_key.compare("flags") == 0 ||
+          omap_key.compare("snap_seq") == 0 ) {
+        unsigned long long val;
```

Comment:
this does not look right. please consult `cls_rbd.cc` for the correct types of these metadata. and accompany with comment documenting why we have these exceptions.